test/unit/core/Object3D.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @author simonThiele / https://github.com/simonThiele
+ */
+
+module( "Object3D" );
+
+test( "getWorldScale", function() {
+  var obj = new THREE.Object3D();
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(0, 0, 0);
```

Comment:
three.js does not support zero-determinant transforms. Most of your tests have a zero scale in at least one dimension. We only need to test what is supported. 

However, if you would like to improve the library so zero-determinant transforms work, that might be useful. I am not sure if there is a compelling reason to do so, however...